internal/xds/resolver/serviceconfig.go

```diff
@@ -363,6 +324,39 @@ func (cs *configSelector) stop() {
 	}
 }
 
+// newInterceptor builds a chain of client interceptors for the given filters.
+// The filter config override maps contain overrides from the route, cluster,
+// and virtual host respectively. The cluster override has the highest priority,
+// followed by the route override, and finally the virtual host override.
+func newInterceptor(filters []xdsresource.HTTPFilter, cluster, route, virtualHost map[string]httpfilter.FilterConfig) (iresolver.ClientInterceptor, error) {
+	if len(filters) == 0 {
+		return nil, nil
+	}
+	interceptors := make([]iresolver.ClientInterceptor, 0, len(filters))
+	for _, filter := range filters {
+		override := cluster[filter.Name]
+		if override == nil {
+			override = route[filter.Name]
+		}
+		if override == nil {
+			override = virtualHost[filter.Name]
+		}
+		ib, ok := filter.Filter.(httpfilter.ClientInterceptorBuilder)
+		if !ok {
+			// Should not happen if it passed xdsClient validation.
+			panic(fmt.Sprintf("filter %q does not support use in client", filter.Name))
```

Comment:
What do you think about continuing to return an error here? The method signature already allows for it, so callers presumably have error handling in place. Panicking to crash the process feels like a worse alternative, especially if this is a recoverable error.